crawl-ref/source/mon-place.cc

```diff
@@ -391,26 +324,22 @@ void spawn_random_monsters()
     if (!x_chance_in_y(5, rate))
         return;
 
-    // Place normal dungeon monsters, but not in player LOS. Don't generate orb
-    // spawns in Abyss to show some mercy to players that get banished there on
-    // the orb run.
-    if (player_in_connected_branch()
-        || (player_on_orb_run() && !player_in_branch(BRANCH_ABYSS)))
+    // Orb spawns. Don't generate orb spawns in Abyss to show some mercy to
+    // players that get banished there on the orb run.
+    if (player_on_orb_run() && !player_in_branch(BRANCH_ABYSS))
     {
         dprf(DIAG_MONPLACE, "Placing monster, rate: %d, turns here: %d",
              rate, env.turns_on_level);
-        proximity_type prox = (one_chance_in(10) ? PROX_NEAR_STAIRS
-                                                 : PROX_AWAY_FROM_PLAYER);
+        proximity_type prox = PROX_AWAY_FROM_PLAYER;
```

Comment:
This `PROX_AWAY_FROM_PLAYER` is never used: on the next line you always overwrite it either with `PROX_CLOSE_TO_PLAYER` or `PROX_ANYWHERE`.

Reply:
Yep, looks like a bit of code only relevant for the now-removed PROX_NEAR_STAIRS got left in, so I'll remove this line.